ModelReplacementAPI/Modules/MoreCompanyCosmeticManager.cs

```diff
@@ -65,71 +62,62 @@ public enum CosmeticType2
+            }
         }
 
         private void DangerousRenderCosmetics(bool useAvatarTransforms)
         {
             RefreshCosmetics();
 
-            var application = controller.gameObject.GetComponentInChildren<CosmeticApplication>();
-            if(application == null) { return; }
+            CosmeticApplication application = controller.gameObject.GetComponentInChildren<CosmeticApplication>();
+            if (application == null) 
+            { 
+                return; 
+            }
+
             if (useAvatarTransforms)
             {
                 foreach (CosmeticInstance cosmeticInstance in application.spawnedCosmetics)
                 {
-                    Transform transform = null;
-                    switch (cosmeticInstance.cosmeticType)
+                    Transform transform = GetCosmeticTransform(cosmeticInstance.cosmeticType);
+                    if (transform != null)
                     {
-                        case CosmeticType.HAT:
-                            transform = cosmeticTransformPairs[CosmeticType2.HAT];
-                            break;
-                        case CosmeticType.CHEST:
-                            transform = cosmeticTransformPairs[CosmeticType2.CHEST];
-                            break;
-                        case CosmeticType.R_LOWER_ARM:
-                            transform = cosmeticTransformPairs[CosmeticType2.R_LOWER_ARM];
-                            break;
-                        case CosmeticType.HIP:
-                            transform = cosmeticTransformPairs[CosmeticType2.HIP];
-                            break;
-                        case CosmeticType.L_SHIN:
-                            transform = cosmeticTransformPairs[CosmeticType2.L_SHIN];
-                            break;
-                        case CosmeticType.R_SHIN:
-                            transform = cosmeticTransformPairs[CosmeticType2.R_SHIN];
-                            break;
+                        ApplyTransformToCosmeticInstance(cosmeticInstance, transform);
                     }
-                    cosmeticInstance.transform.parent = null;
-                    cosmeticInstance.transform.SetPositionAndRotation(transform.position, transform.rotation);
-                    SetAllChildrenLayer(cosmeticInstance.transform, ViewStateManager.modelLayer);
                 }
             }
             else
             {
                 application.RefreshAllCosmeticPositions();
             }
+        }
+
+        private Transform GetCosmeticTransform(CosmeticType cosmeticType)
+        {
+            return cosmeticTransformPairs.TryGetValue((CosmeticType2)cosmeticType, out Transform transform) ? transform : throw new Exception($"Could not find the cosmetic transform");
+        }
 
+        private void ApplyTransformToCosmeticInstance(CosmeticInstance cosmeticInstance, Transform transform)
```

Comment:
Note that I eventually intend to cache a `Dictionary<Transform,Transform>` that relates each cosmetic instance transform to the corresponding model transform to bypass this. I have no issue in passing it now, but it will be modified in the future for performance reasons.